airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultDiscoverCatalogWorker.java

```diff
@@ -82,16 +89,14 @@ public ConnectorJobOutput run(final StandardDiscoverCatalogInput discoverSchemaI
           throw new WorkerException("Integration failed to output a catalog struct.");
         }
 
-        // This message is sent through temporal that utilizes GRPC as is set to limit messages to 4mb.
-        // We fail fast here so users will not have to wait the 10 minutes before a timeout error occurs.
-        if (catalog.get().toString().length() > TEMPORAL_MESSAGE_LIMIT_MB) {
-          throw new WorkerException("Output a catalog struct bigger than 4mb. Larger than grpc max message limit.");
-        }
-
-        return new ConnectorJobOutput().withOutputType(OutputType.DISCOVER_CATALOG).withDiscoverCatalog(catalog.get());
+        final UUID catalogId =
+            configRepository.writeActorCatalogFetchEvent(catalog.get(), UUID.fromString(discoverSchemaInput.getSourceId()),
```

Comment:
This should be moved to an api call in the future. Can we quickly write up an issue so we don't forget?